File: server/TWS Foundation/TWS Customer/Services/Business/TrucksExternalsService.cs


using CSM_Foundation.Database.Entity.Models;
using CSM_Foundation.Database.Models.Out;

using Microsoft.EntityFrameworkCore;

using TWS_Business.Depots;
using TWS_Business.Entities;

using TWS_Customer.Services.Interfaces;

namespace TWS_Customer.Services.Business;
public class TrucksExternalsService : ITrucksExternalsService {
    private readonly TrucksExternalsDepot TrucksExternals;

    public TrucksExternalsService(TrucksExternalsDepot trucksExternals) {
        TrucksExternals = trucksExternals;
    }

    private IQueryable<TruckExternal> Include(IQueryable<TruckExternal> query) {
        return query
            .Include(t => t.Common)
                .ThenInclude(t => t!.Situation)

            .Include(t => t.Common)
                .ThenInclude(t => t!.Location)
                    .ThenInclude(t => t!.Address);
    }

    public async Task<SetViewOut<TruckExternal>> View(SetViewOptions<TruckExternal> Options) {
        return await TrucksExternals.View(Options, Include);
    }

    public async Task<SetBatchOut<TruckExternal>> Create(TruckExternal[] trucks) {
        return await TrucksExternals.Create(trucks);
    }

    public async Task<EntityUpdateOut<TruckExternal>> Update(TruckExternal Truck) {
        return await TrucksExternals.Update(Truck, Include);
    }
}

File: client/tws_foundation_client/lib/src/services/business/trucks_externals_service.dart
import 'package:csm_client/csm_client.dart';
import 'package:tws_foundation_client/tws_foundation_client.dart';

final class TrucksExternalsService extends TrucksExternalsServiceBase {
  TrucksExternalsService(
    CSMUri host, {
    Client? client,
  }) : super(
          host,
          'TrucksExternals',
          client: client,
        );
        
  @override
  Effect<SetViewOut<TruckExternal>> view(SetViewOptions<TruckExternal> options, String auth) async {
    CSMActEffect actEffect = await twsPost('view', options, auth: auth);
    return MainResolver<SetViewOut<TruckExternal>>(actEffect);
  }

  @override
  Effect<SetBatchOut<TruckExternal>> create(List<TruckExternal> trucks, String auth) async {
    CSMActEffect actEffect = await twsPostList('create', trucks, auth: auth);
    return MainResolver<SetBatchOut<TruckExternal>>(actEffect);
  }

  @override
  Effect<RecordUpdateOut<TruckExternal>> update(TruckExternal truck, String auth) async {
    CSMActEffect actEffect = await twsPost('update', truck, auth: auth);
    return MainResolver<RecordUpdateOut<TruckExternal>>(actEffect);
  }
}
      